Realm/RLMProperty.h

```diff
@@ -92,4 +97,17 @@ RLM_ASSUME_NONNULL_BEGIN
 
 @end
 
+
+/**
+ This class describes a specific property on a given class.
+ */
+@interface RLMPropertyDescriptor : NSObject
+
++ (instancetype)descriptorWithClass:(Class)objectClass propertyName:(NSString *)propertyName;
```

Comment:
Have we ever considered using selectors for the property name? That would give users a bit more compile-time safety when using Realm Objective-C from Swift / Realm Swift.